gensim/models/base_any2vec.py

```diff
@@ -459,25 +465,36 @@ def __str__(self):
             self.__class__.__name__, len(self.wv.index2word), self.vector_size, self.alpha
         )
 
-    def build_vocab(self, sentences, update=False, progress_per=10000, keep_raw_vocab=False, trim_rule=None, **kwargs):
+    def build_vocab(self, sentences, multistream=False, workers=None, update=False, progress_per=10000,
+                    keep_raw_vocab=False, trim_rule=None, **kwargs):
         """Build vocabulary from a sequence of sentences (can be a once-only generator stream).
         Each sentence is a iterable of iterables (can simply be a list of unicode strings too).
 
         Parameters
         ----------
-        sentences : iterable of iterables
+        sentences : {iterable of iterables, list or tuple of iterable of iterables}
             The `sentences` iterable can be simply a list of lists of tokens, but for larger corpora,
             consider an iterable that streams the sentences directly from disk/network.
             See :class:`~gensim.models.word2vec.BrownCorpus`, :class:`~gensim.models.word2vec.Text8Corpus`
             or :class:`~gensim.models.word2vec.LineSentence` in :mod:`~gensim.models.word2vec` module for such examples.
+            If `multistream=True`, `sentences` must be list or tuple of input streams described above.
+        multistream : bool
+            If True, use `sentences` as list of input streams and speed up vocab building by parallelization
+            with `min(len(sentences), self.workers)` processes. This option can lead up to 2.5x reduction
+            in vocabulary building time.
+        workers : int
```

Comment:
maybe better use different naming like `multistream_workers` (for avoiding potential collision by parameter names) or this has no sense @persiyanov?

Reply:
I think there is no point in this. In most cases, by setting `workers` parameter user means the same degree of parallelization for both scan vocab (multiprocessing) and training the model (multithreading), IMO